File: 2.Algorithm/03.DFS, BFS/타겟넘버/타겟넘버.py
from collections import deque

def solution(numbers, target):
    i = 1
    q = deque()
    q.append(numbers[0])
    q.append(-numbers[0])
    while i < len(numbers):
        temp = q.popleft()
        q.append(temp+numbers[i])
        q.append(temp-numbers[i])
        if len(q) == 2 ** (i+1):
            i += 1
    return q.count(target)

File: 2.Algorithm/03.DFS, BFS/타겟넘버/타겟넘버.java
class Solution {

    public int solution(int[] numbers, int target) {
        int answer = 0;
        answer = re(numbers, target, 0, 0);
        return answer;
    }

    int re(int[] numbers, int target, int sum, int n){
        if(numbers.length == n){
            if(sum == target){
                return 1;
            }
            return 0;
        }
        return re(numbers, target, sum + numbers[n], n + 1) + re(numbers, target, sum - numbers[n], n + 1); 
    }
}